src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -175,6 +176,44 @@ private void AddExceptions(ISymbol symbol)
                 }
             }
 
+            protected abstract void AddCaptures(ISymbol symbol);
+
+            protected void AddCaptures(SyntaxNode syntax)
+            {
+                var semanticModel = GetSemanticModel(syntax.SyntaxTree);
+                if (semanticModel.IsSpeculativeSemanticModel)
+                {
+                    // Speculative semantic models cannot do region analysis
+                    return;
+                }
+
+                var analysis = semanticModel.AnalyzeDataFlow(syntax);
+                var captures = analysis.CapturedInside;
+                if (!captures.IsEmpty)
+                {
+                    var parts = new List<SymbolDisplayPart>();
+                    parts.Add(new SymbolDisplayPart(kind: SymbolDisplayPartKind.Text, symbol: null, text: $"\r\n{WorkspacesResources.Captures_colon}"));
+                    bool first = true;
+                    foreach (var captured in captures)
+                    {
+                        if (!first)
+                        {
+                            parts.AddRange(Punctuation(","));
+                        }
+
+                        parts.AddRange(Space(count: 1));
+                        parts.AddRange(ToMinimalDisplayParts(captured, s_formatForCaptures));
+                        first = false;
+                    }
+
+                    AddToGroup(SymbolDescriptionGroups.Captures, parts);
+                }
+            }
+
+            private static readonly SymbolDisplayFormat s_formatForCaptures = SymbolDisplayFormat.MinimallyQualifiedFormat
+                .RemoveLocalOptions(SymbolDisplayLocalOptions.IncludeType)
+                .RemoveParameterOptions(SymbolDisplayParameterOptions.IncludeType);
```

Comment:
Are you basically just writing the name of the local/parameter?  If you're removing the type, it's unclear why use symboldisplay or formats at all.

Reply:
I think I addressed this elsewhere, but by understanding is that the symbol display handles not just strings, but also colorization. I'd rather rely on that machinery (and fix that shared code as appropriate).

---
In reply to: [160014346](https://github.com/dotnet/roslyn/pull/23970#discussion_r160014346) [](ancestors = 160014346)

Reply:
Ah.  ok.